internal/filesystem/ipfs/ufs.go

```diff
@@ -0,0 +1,63 @@
+package ipfs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	fserrors "github.com/djdv/go-filesystem-utils/internal/filesystem/errors"
+	"github.com/djdv/go-filesystem-utils/internal/generic"
+	files "github.com/ipfs/boxo/files"
+	unixfsfile "github.com/ipfs/boxo/ipld/unixfs/file"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+type (
+	ufsFile struct {
+		files.File
+		cancel context.CancelFunc
+		info   nodeInfo
+	}
+)
```

Comment:
Mentioned by @aschmahmann in offline review.

the `files.File` we embed here, is derived from `unixfsfile.NewUnixfsFile` which takes in a context already, and has simillar semantics internally (it cancels on `Close`).

Our wrapper could probably be reduced to remove some redundancy, instead just relying on the embedded methods and behaviour being hoisted up.

This might be true for other types too. But requires peeking into the implementations they wrap.

Reply:
Issue filed: https://github.com/djdv/go-filesystem-utils/issues/33